simulator/gdb/sim_wrapper.cpp

```diff
@@ -0,0 +1,78 @@
+/**
+ * sim_wrapper.cpp - Functional simulator wrappers for GDB
```

Comment:
At first I tried to do it in one file. Suddenly, one of GDB headers contains a variable named "xor", and C++ compiler treats it as the xor operator, so it can't be compiled with C++. Actually, this is the only reason why I created a separate C file.

Reply:
Sounds like a bug: GDB code base has a lot of C++ code, so such stuff should be purged. Could you please point to the definition of that "xor" variable?

Reply:
https://github.com/bminor/binutils-gdb/blob/75cbc781e371279f4403045be93b07fd8fe7fde5/sim/common/sim-core.h#L92

Yes, GDB uses C++ and even Python, however, all simulator examples are written in C. I assume simulator interfaces are supposed to be used in C.

Reply:
I advise you to rename it to 'byte_xor' and submit a patch (see [CONTRIBUTING.md](
http://sourceware.org/git/gitweb.cgi?p=binutils-gdb.git;a=blob;f=gdb/CONTRIBUTE;hb=HEAD)).

Reply:
And, for sure, fix it locally.